src/commands/monitor.rs

```diff
@@ -511,10 +514,14 @@ async fn submit_and_watch_solution<T: MinerConfig + Send + Sync + 'static>(
 		.create_signed_with_nonce(&tx, &*signer, nonce, ExtrinsicParams::default())?;
 
 	if dry_run {
-		match api.rpc().dry_run(xt.encoded(), None).await? {
-			Ok(()) => (),
-			Err(e) => return Err(Error::TransactionRejected(format!("{e:?}"))),
-		};
+		let dry_run_bytes = api.rpc().dry_run(xt.encoded(), Some(block_hash)).await?;
+
+		match dry_run_bytes.into_dry_run_result(&api.metadata())? {
```

Comment:
/cc @jsdw I reckon that this is expensive and I didn't find a better way, how slow is it to read the RwLock and clone the metadata?

Reply:
Ooh I forgot about that!  Yeah, I wouldn't worry about it; this is no less efficient than any other approach :)